File: Signiture-Backend/delivery/db/src/main/java/pcy/study/db/userorder/UserOrderRepository.java
package pcy.study.db.userorder;

import org.springframework.data.jpa.repository.JpaRepository;
import pcy.study.db.userorder.enums.UserOrderStatus;

import java.util.List;
import java.util.Optional;

public interface UserOrderRepository extends JpaRepository<UserOrder, Long> {

    List<UserOrder> findAllByUserIdAndStatusOrderByIdDesc(Long userId, UserOrderStatus status);

    List<UserOrder> findAllByUserIdAndStatusInOrderByIdDesc(Long userId, List<UserOrderStatus> status);

    Optional<UserOrder> findFirstByIdAndUserIdAndStatus(Long id, Long userId, UserOrderStatus status);
}

File: Signiture-Backend/kotlin/delivery/db/src/main/kotlin/pcy/study/db/userorder/UserOrderRepository.kt
package pcy.study.db.userorder

import org.springframework.data.jpa.repository.JpaRepository
import pcy.study.db.userorder.enums.UserOrderStatus

interface UserOrderRepository : JpaRepository<UserOrder, Long> {

    fun findAllByUserIdAndStatusOrderByIdDesc(
        userId: Long?,
        status: UserOrderStatus?
    ): List<UserOrder>

    fun findAllByUserIdAndStatusInOrderByIdDesc(
        userId: Long?,
        status: List<UserOrderStatus>
    ): List<UserOrder>

    fun findFirstByIdAndUserId(
        id: Long?,
        userId: Long?
    ): UserOrder?
}
